packages/discovery-provider/src/queries/get_follow_intersection_users.py

```diff
@@ -1,51 +1,41 @@
-from src.models.social.follow import Follow
-from src.models.users.user import User
-from src.queries import response_name_constants
-from src.queries.query_helpers import (
-    get_current_user_id,
-    paginate_query,
-    populate_user_metadata,
-)
-from src.utils import helpers
+from sqlalchemy.sql import text
+
+from src.queries.get_unpopulated_users import get_unpopulated_users
+from src.queries.query_helpers import populate_user_metadata
 from src.utils.db_session import get_db_read_replica
 
+sql = text(
+    """
+select
+  x.follower_user_id
+from follows x
+join aggregate_user au on x.follower_user_id = au.user_id
+join follows me
+  on me.follower_user_id = :my_id
+  and me.followee_user_id = x.follower_user_id
+  and me.is_delete = false
+where x.followee_user_id = :other_user_id
+  and x.is_delete = false
+order by follower_count desc
+limit :limit
+offset :offset
+"""
+)
 
-def get_follow_intersection_users(args):
-    followee_user_id = args.get("followee_user_id")
-    follower_user_id = args.get("follower_user_id")
 
+def get_follow_intersection_users(args):
     users = []
+    my_id = args.get("my_id")
```

Comment:
nit: The canonical pattern for this right now is `user_id` (and sometimes `current_user_id`) and it would be nice to not mix patterns. The reason is that it's not always _my_ followers. It's the followers for a target user. Gets even more confusing in manager mode where the UI mostly looks like you are "me", but you're still calling with a different signing address, so everything is recontextualized to that user.

I don't mind if inside this query, the incoming argument gets renamed for convenience in the SQL. But I would prefer we don't start changing queries to use different names for the contextual user id parameter. :pray:

Reply:
i missed this rename, good catch. yeah i think keep followee/follower for now tbh. If we _really_ want to change them, `followed_user_id` and `follower_user_id` I think would be better?